contracts/registries/LicenseRegistry.sol

```diff
@@ -344,18 +345,27 @@ contract LicenseRegistry is ILicenseRegistry, AccessManagedUpgradeable, UUPSUpgr
         uint256 index
     ) external view returns (address licenseTemplate, uint256 licenseTermsId) {
         LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
-        if (index >= $.attachedLicenseTerms[ipId].length()) {
-            revert Errors.LicenseRegistry__IndexOutOfBounds(ipId, index, $.attachedLicenseTerms[ipId].length());
+        // consider the default license terms is attached to IP as the last one
+        uint256 length = $.attachedLicenseTerms[ipId].length();
+        if (index < length) {
+            licenseTemplate = $.licenseTemplates[ipId];
+            licenseTermsId = $.attachedLicenseTerms[ipId].at(index);
+            // consider the default license terms is attached to IP as the last one
+        } else if (index == length && $.defaultLicenseTemplate != address(0)) {
+            licenseTemplate = $.defaultLicenseTemplate;
+            licenseTermsId = $.defaultLicenseTermsId;
+        } else {
+            length += ($.defaultLicenseTemplate == address(0) ? 0 : 1);
```

Comment:
One question: 
Would it be clearer or more explicit to use the following instead?  

```solidity
($.defaultLicenseTemplate == address(pilTemplate) ? 1 : 0)
```
Given that the default license template is intended to use `pilTemplate`, would this make the logic more readable?

Reply:
in theory, the defaultLicenseTemplate can be any license template, not have to pil license template.